tests/MomentoTaggedCacheTest.php

```diff
@@ -0,0 +1,124 @@
+    {
+        $key = uniqid();
+        $value = uniqid();
+        $putResult = Cache::tags([uniqid()])->put($key, $value, 5);
+        $this->assertTrue($putResult, "True was expected but received ${putResult}");
+        $getResult = Cache::tags([])->get($key);
+        $this->assertFalse($getResult, "False was expected but received ${putResult}");
+    }
+
+    /**
+     * @covers \Momento\Cache\MomentoTaggedCache
+     */
+    public function testTaggedPutMultipleItems_WithOneTag_HappyPath()
+    {
+        $key1 = uniqid();
+        $value1 = uniqid();
+        $key2 = uniqid();
+        $value2 = uniqid();
+        $tag1 = uniqid();
+        $putResult = Cache::tags([$tag1])->put($key1, $value1, 5);
+        $this->assertTrue($putResult, "True was expected but received ${putResult}");
+        $putResult = Cache::tags([$tag1])->put($key2, $value2, 5);
+        $this->assertTrue($putResult, "True was expected but received ${putResult}");
+        $getResult = Cache::tags([$tag1])->get($key1);
+        $this->assertEquals($value1, $getResult, "${value1} was expected but received ${getResult}");
+        $getResult = Cache::tags([$tag1])->get($key2);
+        $this->assertEquals($value2, $getResult, "${value2} was expected but received ${getResult}");
+    }
+
+    /**
+     * @covers \Momento\Cache\MomentoTaggedCache
+     * @covers \Momento\Cache\MomentoStore
+     */
+    public function testTaggedPut_RegularGet_IsNull()
+    {
+        $key = uniqid();
+        $value = uniqid();
+        $tag = uniqid();
+        $putResult = Cache::tags([$tag])->put($key, $value, 5);
+        $this->assertTrue($putResult, "True was expected but received ${putResult}");
+        $getResult = Cache::get($key);
+        $this->assertNull($getResult, "null was expected but received ${getResult}");
+    }
+
+    /**
+     * @covers \Momento\Cache\MomentoTaggedCache
+     */
+    public function testTaggedPut_WithWrongOrderTaggedGet_IsNull()
+    {
+        $key = uniqid();
+        $value = uniqid();
+        $tag1 = uniqid();
+        $tag2 = uniqid();
+        $putResult = Cache::tags([$tag1, $tag2])->put($key, $value, 5);
+        $this->assertTrue($putResult, "True was expected but received ${putResult}");
+        $getResult = Cache::tags([$tag2, $tag1])->get($key);
+        $this->assertNull($getResult, "null was expected but received ${getResult}");
+    }
+
+    /**
+     * @covers \Momento\Cache\MomentoTaggedCache
+     */
+    public function testTaggedPutWithTwoTags_TaggedGetWithOnlyOneTag_IsNull()
+    {
+        $key = uniqid();
+        $value = uniqid();
+        $tag1 = uniqid();
+        $tag2 = uniqid();
+        $putResult = Cache::tags([$tag1, $tag2])->put($key, $value, 5);
+        $this->assertTrue($putResult, "True was expected but received ${putResult}");
+        $getResult = Cache::tags([$tag1])->get($key);
+        $this->assertNull($getResult, "null was expected but received ${getResult}");
+    }
+}
```

Comment:
I have not added `flush` to `MomentoTaggedCache` so it will probably throw an exception that does not make sense to users. 
I will add the method and throw an `UnknownError` saying that `flush` is not currently supported until we implement it.

Reply:
start by writing a test to see what it does, no need to guess

Reply:
It turns out that if I call `flush` on tagged cache, I get `Momento\Cache\Errors\UnknownError: forever operations is currently not supported.`.
Since `flush` is not implemented in `MomentoTaggedCache`, it goes to `forever` operation which throws the above error.
The below are the stack trace:
```
/Users/erikatharp/laravel-cache/src/MomentoStore.php:94
/Users/erikatharp/laravel-cache/vendor/laravel/framework/src/Illuminate/Cache/TagSet.php:54
/Users/erikatharp/laravel-cache/vendor/laravel/framework/src/Illuminate/Cache/TagSet.php:43
/Users/erikatharp/laravel-cache/vendor/laravel/framework/src/Illuminate/Cache/TaggedCache.php:81
```

Reply:
The error thrown for calling `flush` is confusing, so I will throw an error with this message:
```
 flush operations is currently not supported.
```

Reply:
The error thrown for calling `flush` is confusing, so I will throw an error with this message:
```
 flush operations is currently not supported.
```